### PR TITLE
Fix/language detection reproducibility

### LINKED ISSUE #1229: Improve Reproducibility and Prevent Data Leakage in Train-Test Pipeline
**Describe the bug**
I noticed two related issues while going through LanguageDetectionModel.ipynb in the current preprocessing and train-test pipeline that could affect model reliability and evaluation quality.
1. Non-Reproducible Train/Test Split- The current train_test_split() call does not use a fixed random_state, so the train/test data changes on every execution. This leads to inconsistent accuracy scores and makes experiments difficult to reproduce.
2. Potential Data Leakage During Vocabulary Creation- Currently, the vocabulary and count vectors are created before performing the train-test split. This can introduce data leakage because the feature extraction process indirectly uses information from the test set.

**To Reproduce**
1. Run the notebook multiple times without modifying the code.
2. Observe that the train-test split changes on every execution because random_state is not specified.
3. Compare the model accuracy across runs.
4. Notice inconsistent evaluation results.
Additionally, observe that vocabulary creation and count vector generation occur before the train-test split. This causes the feature extraction process to use information from the entire dataset, including the test set.

**Expected behavior**
The train-test split should remain consistent across runs for reproducible evaluation.
Feature extraction (vocabulary/vectorizer fitting) should be performed only on the training data to avoid data leakage.
The test set should remain completely unseen during training and preprocessing stages.

I would like to work on this issue and contribute a fix for it. Kindly assign this issue to me under GSSoC'26.

### PR DESCRIPTION
**Pull request by vaibhavis9024**
### Issues Fixed
Removed data leakage by performing train_test_split before building the vocabulary/count vectors
Built vocabulary using only training data

### Fixed incorrect append statement:

count_vec.append(X_train) → count_vec.append(count_arr)
Reordered cells so Kernel → Restart & Run All executes without errors
Added safe handling for unseen test words using if w in word_dic
### Result
Clean sequential notebook execution
No NameError or vocabulary issues
Reproducible accuracy across runs (97.16%)

### CODE REVIEW

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]
Image

many errors @vaibhavis9024 
@vaibhavis9024 @vaibhavis9024 fix it

**Comment by Sibam-Paul** [comment]
noActivity of 2w of review closing
